scripts/add_electricity.py

```diff
@@ -428,20 +428,33 @@ def attach_hydro(n, costs, ppl):
     inflow_idx = ror.index.union(hydro.index)
     if not inflow_idx.empty:
         with xr.open_dataarray(snakemake.input.profile_hydro) as inflow:
-            inflow_stations = pd.Index(bus_id[inflow_idx])
-            missing_c = inflow_stations.unique().difference(inflow.indexes["plant"])
-            assert missing_c.empty, (
-                f"'{snakemake.input.profile_hydro}' is missing "
-                f"inflow time-series for at least one bus: {', '.join(missing_c)}"
+            inflow_stations = bus_id[inflow_idx]
+            missing_c = pd.Index(inflow_stations.unique()).difference(
+                inflow.indexes["plant"]
             )
+            intersection_c = inflow.indexes["plant"].intersection(inflow_stations)
+
+            # if missing time series are found, notify the user and exclude missing hydro plants
+            if not missing_c.empty:
+                idxs_to_keep = inflow_stations[
+                    inflow_stations.isin(intersection_c)
+                ].index
+                ror = ror.loc[ror.index.intersection(idxs_to_keep)]
+                hydro = hydro.loc[hydro.index.intersection(idxs_to_keep)]
```

Comment:
A very nice and clean solution by its essence. What has been not easy to understand for me when reading the code are the naming conventions. A couple of suggestions:
1) Is it possible to clarify in the names what does `"_c"` imply in `missing_c` and `intersection_c`?
2) Usually we refer generators as power plants, while the`"stations"` is used in variables names related to a power plants dataset, while `stations` might have implications with meteorological measurements. Would it be worth to revise naming a bit?

Reply:
Revised:
- _c -> _plants
- stations -> buses